packages/node_modules/@ciscospark/widget-recents/src/container.js

```diff
@@ -160,15 +173,20 @@ export class RecentsWidget extends Component {
         }
       }
 
-      // Listen for incoming calls
-      const hasCheckedWebRTCSupport = media.getIn(['webRTC', 'hasCheckedSupport']);
-      const isWebRTCSupported = media.getIn(['webRTC', 'isSupported']);
+      if (widgetStatus.hasFetchedRecentSpaces) {
+        // Wait to listen for incoming calls until spaces have loaded
+        const hasCheckedWebRTCSupport = media.getIn(['webRTC', 'hasCheckedSupport']);
+        const isWebRTCSupported = media.getIn(['webRTC', 'isSupported']);
 
-      if (!hasCheckedWebRTCSupport) {
-        props.checkWebRTCSupport(sparkInstance);
-      }
-      if (isWebRTCSupported && !media.getIn(['status', 'isListening'])) {
-        props.listenForIncomingCalls(sparkInstance);
+        if (!hasCheckedWebRTCSupport) {
+          props.checkWebRTCSupport(sparkInstance);
+        }
+        if (isWebRTCSupported && !media.getIn(['status', 'isListening'])) {
+          RecentsWidget.getCurrentCalls(props);
+          props.listenForCalls(sparkInstance);
+        }
```

Comment:
Could we move to the new mercury and media enhancer to get these checks? Would clean up the container a lot.